### PR TITLE
Positive numerical tasks

### PR DESCRIPTION
**Pull request by paxcema**
Avoid negative predictions for positive numerical targets by:
- Detecting positive numerical domains based on histogram
- Sending corresponding flag to Lightwood numerical encoder (see [this PR](https://github.com/mindsdb/lightwood/pull/318))

The PR also adds a small test to detect if this is working as intended.

### CODE REVIEW

File: mindsdb_native/libs/phases/data_analyzer/data_analyzer.py [hunk 1]
```diff
@@ -299,18 +299,22 @@ def run(self, input_data):
...
+                            outlier_values=outliers,
+                            hist_x=histogram['x'],
+                            hist_y=histogram['y'],
+                            percentage_buckets=percentage_buckets,
+                            col_stats=stats_v2[col_name]
+                        ),
+                        'description': """Potential outliers can be thought as the "extremes", i.e., data points that are far from the center of mass (mean/median/interquartile range) of the data."""
+                    }
+                    # specify positive numerical domain
+                    if stats_v2[col_name]['histogram']['x'][0] >= 0:
```
Comment (maximlopin) [inline mindsdb_native/libs/phases/data_analyzer/data_analyzer.py:315]: is it assumed that `histogram['x']` is sorted?

Comment (paxcema) [reply mindsdb_native/libs/phases/data_analyzer/data_analyzer.py:315]: Yes. This is only for the numerical data type, and from what I can gather, `np.histogram` returns sorted bins.

File: mindsdb_native/libs/controllers/transaction.py [hunk 2]
```diff
@@ -359,7 +359,12 @@ def run(self):
                                 diff = sample[1, idx] - sample[0, idx]
                                 if diff <= tolerance:
                                     output_data[f'{predicted_col}_confidence'][sample_idx] = significance
-                                    output_data[f'{predicted_col}_confidence_range'][sample_idx] = list(sample[:, idx])
+                                    conf_range = list(sample[:, idx])
+
+                                    # for positive numerical domains
+                                    if self.lmd['stats_v2'][predicted_col].get('positive_domain', False):
+                                        conf_range[0] = max(0, conf_range[0])
```
Comment (maximlopin) [inline mindsdb_native/libs/controllers/transaction.py:366]: what if `conf_range` at indices greater than `0` has negative values?

Comment (paxcema) [reply mindsdb_native/libs/controllers/transaction.py:366]: ~~Good catch, that _could_ happen~~

Comment (paxcema) [reply mindsdb_native/libs/controllers/transaction.py:366]: Scratch that, it cannot happen. We get a guarantee from Lightwood that predictions are `>= 0` for positive domains, and the sigma from the conformal predictor can never be negative. Still, I'll add a test for this.

**Review by maximlopin — APPROVED** [review]

**Pull request merged**